src/WinReg.jl

```diff
@@ -1,130 +1,31 @@
-const KEY_WOW64_64KEY         = 0x00100  # Indicates that an application on 64-bit Windows should operate on the 64-bit registry view. This flag is ignored by 32-bit Windows. For more information, see Accessing an Alternate Registry View.
-# This flag must be combined using the OR operator with the other flags in this table that either query or access registry values.
-# Windows 2000:  This flag is not supported.
-
-const KEY_WRITE               = 0x20006  # Combines the STANDARD_RIGHTS_WRITE, KEY_SET_VALUE, and KEY_CREATE_SUB_KEY access rights.
-
-function openkey(base::UInt32, path::AbstractString, accessmask::UInt32=KEY_READ)
-    keyref = Ref{UInt32}()
-    ret = ccall((:RegOpenKeyExW, "advapi32"),
-                stdcall, Clong,
-                (UInt32, Cwstring, UInt32, UInt32, Ref{UInt32}),
-                base, path, 0, accessmask, keyref)
-    if ret != 0
-        error("Could not open registry key")
-    end
-    keyref[]
-end
-
-function querykey(key::UInt32, valuename::AbstractString)
-    dwSize = Ref{UInt32}()
-    dwDataType = Ref{UInt32}()
-
-    ret = ccall((:RegQueryValueExW, "advapi32"),
-                stdcall, Clong,
-                (UInt32, Cwstring, Ptr{UInt32},
-                 Ref{UInt32}, Ptr{UInt8}, Ref{UInt32}),
-                key, valuename, C_NULL,
-                dwDataType, C_NULL, dwSize)
-    if ret != 0
-        error("Could not find registry value name")
-    end
-
-    data = Array(UInt8, dwSize[])
-    ret = ccall((:RegQueryValueExW, "advapi32"),
-                stdcall, Clong,
-                (UInt32, Cwstring, Ptr{UInt32},
-                 Ptr{UInt32}, Ptr{UInt8}, Ref{UInt32}),
-                key, valuename, C_NULL,
-                C_NULL, data, dwSize)
-    if ret != 0
-        error("Could not retrieve registry data")
-    end
-
-    if dwDataType[] == REG_SZ || dwDataType[] == REG_EXPAND_SZ
-        data_wstr = reinterpret(Cwchar_t,data)
-
-        # string may or may not be null-terminated
-        if data_wstr[end] == 0
-            pop!(data_wstr)
-        end
-
-        @static if isdefined(Base,:transcode)
-            return String(transcode(UInt8,data_wstr))
-        else
-            return bytestring(wstring(data_wstr))
-        end
-    elseif dwDataType[] == REG_DWORD
-        return reinterpret(Int32,data)[]
-    elseif dwDataType[] == REG_QWORD
-        return reinterpret(Int64,data)[]
-    else
-        return data
-    end
-end
-
-function querykey(base::UInt32, path::AbstractString, valuename::AbstractString)
-    key = openkey(base,path)
-    val = querykey(key, valuename)
-    closekey(key)
-    val
-end
-
-function closekey(key::UInt32)
-    ret = ccall((:RegCloseKey, "advapi32"),
-                stdcall, Clong,
-                (UInt32,),
-                key)
-    if ret != 0
-        error("Could not close key")
-    end
-    nothing
+#Julia 0.4.x compatibility
+if VERSION < v"0.5.0"
+	include_string("is_windows() = (@windows ? true : false)")
```

Comment:
Mmmh ...
Originally my version was suitable only for Julia 0.5.0, but If you have in mind some requirement to mantain an updated Julia 0.4.x version, I agree.

Because other specific Julia 0.5.0 code should be re-adapted to Julia 0.4.x, I accept suggestion about best way to do so without inject too much `@compat` code to remove / modify in subsequent Julia releases